include/envoy/upstream/outlier_detection.h

```diff
@@ -34,6 +35,18 @@ class DetectorHostSink {
    * different operations including HTTP, Mongo, Redis, etc.).
    */
   virtual void putResponseTime(std::chrono::milliseconds time) PURE;
+
+  /**
+   * Get the time of last ejection.
+   * @return the last time this host was ejected
+   */
+  virtual SystemTime ejectionTime() PURE;
```

Comment:
Optional<SystemTime>, may not be any prior ejection. This will clean up other code you have. Same below.